types/api.go

```diff
@@ -195,3 +211,124 @@ type Key struct {
 	Idx int    `json:"idx"`
 	Key string `json:"key"`
 }
+
+// VochainStats contains information about the current Vochain statistics and state
+type VochainStats struct {
+	BlockHeight      uint32    `json:"block_height"`
+	EntityCount      int64     `json:"entity_count"`
+	EnvelopeCount    uint64    `json:"envelope_count"`
+	ProcessCount     int64     `json:"process_count"`
+	ValidatorCount   int       `json:"validator_count"`
+	BlockTime        *[5]int32 `json:"block_time"`
+	BlockTimeStamp   int32     `json:"block_time_stamp"`
+	ChainID          string    `json:"chain_id"`
+	GenesisTimeStamp time.Time `json:"genesis_time_stamp"`
+
+	// TransactionHeight int64     `json:"transaction_height"`
+	// AvgTxsPerBlock    float64   `json:"avg_txs_per_block"`
+	// AvgTxsPerMinute   float64   `json:"avg_txs_per_minute"`
+	// The hash of the block with the most txs
+	// MaxBytes          int64  `json:"max_bytes"`
+	// MaxTxsBlockHash   string `json:"max_txs_block_hash"`
+	// MaxTxsBlockHeight int64  `json:"max_txs_block_height"`
+	// The start of the minute with the most txs
+	// MaxTxsMinute    time.Time `json:"max_txs_minute"`
+	// MaxTxsPerBlock  int64     `json:"max_txs_per_block"`
+	// MaxTxsPerMinute int64     `json:"max_txs_per_minute"`
+
+	Syncing bool `json:"syncing"`
+}
+
+// EnvelopePackage contains a VoteEnvelope and auxiliary information for the Envelope api
+type EnvelopePackage struct {
+	Nonce                HexBytes
+	ProcessId            HexBytes
+	VotePackage          []byte
+	Nullifier            HexBytes
+	EncryptionKeyIndexes []uint32
+	Weight               []byte
+	TxIndex              int32
+	Height               uint32
+	TxHash               HexBytes
+}
+
+// EnvelopeMetadata contains vote information for the EnvelopeList api
+type EnvelopeMetadata struct {
+	ProcessId HexBytes
+	Nullifier HexBytes
+	TxIndex   int32
+	Height    uint32
+	TxHash    HexBytes
+}
+
+// TxPackage contains a SignedTx and auxiliary information for the Transaction api
+type TxPackage struct {
+	Tx          []byte
+	BlockHeight uint32
+	Index       int32
+	Hash        HexBytes
+	Signature   HexBytes
+}
+
+// TxMetadata contains tx information for the TransactionList api
+type TxMetadata struct {
+	Type        string
+	BlockHeight uint32
+	Index       int32
+	Hash        HexBytes
+}
+
+// BlockMetadata contains the metadata for a single tendermint block
+type BlockMetadata struct {
```

Comment:
don't we want these types to also have json field tags? otherwise the json keys will be like `FooBar` by default, instead of something more consistent with the existing types like `foo_bar`